### PR TITLE
add fish_indent fixer for fish shell scripts

### LINKED ISSUE #2148: Add support for fish_indent as a fixer
**Name:** fish_indent
**URL:** http://fishshell.com/docs/current/commands.html#fish_indent

The FISH shell ships its own formatter, **fish_indent**, and I think it would work well as an ALE fixer.

I've found one implementation so far of getting it integrated with ALE:

https://github.com/nthapaliya/dotfiles/blob/aef2e112f5207fb63ba1c492b08697fbb2be17fb/home/.config/nvim/init.vim

This project integrates fish_indent with "gq":
https://github.com/dag/vim-fish/

### PR DESCRIPTION
**Pull request by cyyever**
fix #2148

### CODE REVIEW

**Comment by cyyever** [comment]
@w0rp  Any plan to merge or refuse?

**Comment by cyyever** [comment]
@w0rp doc and test are added.

**Comment by cyyever** [comment]
@w0rp Could you merge it?

**Comment by lithammer** [comment]
Not stale, ping @w0rp

File: doc/ale-fish.txt [hunk 1]
```diff
@@ -10,5 +10,22 @@ displaying errors if an error message is not found.
 If ALE is not showing any errors but your file does not run as expected, run
 `fish -n <file.fish>` from the command line.
 
+===============================================================================
+fish_indent                                               *ale-fish-fishindent*
```
[the remark this answers is not in the document]
Comment (cyyever) [reply doc/ale-fish.txt:14]: @MindTooth Many linters have a _, such as clang_format.

**Comment by cyyever** [comment]
@w0rp  Could you consider reopening and merging it?

File: doc/ale-fish.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (MindTooth) [reply doc/ale-fish.txt:14]: Yes.  That’s why I asked why it did not have it.

Maybe I were not clear.  My bad.  I ment for the help index line.  Missing a underline there.

[the remark this answers is not in the document]
Comment (cyyever) [reply doc/ale-fish.txt:14]: @MindTooth fixed it.

**Comment by cyyever** [comment]
@MindTooth  could you merge it?

**Comment by MindTooth** [comment]
> @MindTooth could you merge it?

I’m no maintainer.  My bad if I gave that impression.

**Comment by MindTooth** [comment]
@hsanson Care to take a look at this when you have the time?  I feel this got closed prematurely.

**Comment by hsanson** [comment]
@MindTooth the PR looks good but Github does not allow to re-open it because it was either force pushed or recreated while it was closed.  Is easier if @cyyever creates a new PR from the same branch.

**Comment by cyyever** [comment]
> 
> 
> @MindTooth the PR looks good but Github does not allow to re-open it because it was either force pushed or recreated while it was closed. Is easier if @cyyever creates a new PR from the same branch.

New PR is #3576

**Pull request closed without merging**